### PR TITLE
Textarea component with tests

### LINKED ISSUE #16: Add a textarea component

### PR DESCRIPTION
**Pull request by RizaHKhan**
#16

### CODE REVIEW

File: src/components/atoms/BaseTextArea.atom.vue [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    :disabled="isDisabled"
+    :required="isRequired"
+  ></textarea>
+</template>
+
+<script lang="ts">
+import { defineComponent, SetupContext } from 'vue'
+
+interface Props {
+  id: string
```
Comment (JadRizk) [inline src/components/atoms/BaseTextArea.atom.vue:21]: This is causing a TS error on the `setup` function.

Try this instead: 
```suggestion
id?: string;
```

File: src/components/atoms/BaseTextArea.atom.vue [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  rows?: string
+  modelValue?: string
+  errorMode?: boolean
+  placeholder?: string
+  isDisabled: boolean
+  isRequired: boolean
+}
+
+export default defineComponent({
+  name: 'BaseTextInput',
```
Comment (JadRizk) [inline src/components/atoms/BaseTextArea.atom.vue:33]:
```suggestion
  name: 'BaseTextArea',
```

**Review by JadRizk — APPROVED** [review]

**Review by JadRizk — CHANGES_REQUESTED** [review]
I believe that the issue with the `id` param is causing the Netlify checks to fail. 

Maybe we should be fixing that first.

File: src/components/atoms/BaseTextArea.atom.vue (hunk 2 above)
Comment (RizaHKhan) [reply src/components/atoms/BaseTextArea.atom.vue]: omg silly mistake, sorry.

Corrected.

**Review by JadRizk — APPROVED** [review]
Sounds good 💥

**Pull request merged**